### ISSUE TITLE
Basing calculations only on sensor data fails - needs at least one reading to come from OWM

### ISSUE THREAD

**Issue by MrBurnst** [jeroenterheerdt/HAsmartirrigation#403]
### What happened?

since 2024.6.3, at least one of the measurements for the calculations have to come from OWM, otherwise I'm getting

```
Logger: aiohttp.server
Source: /usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py:421
First occurred: 9:09:22 PM (9 occurrences)
Last logged: 9:09:24 PM

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 543, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 114, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 92, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 83, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 26, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 85, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 242, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 32, in headers_middleware
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/http.py", line 73, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 70, in wrapper
    return await method(view, request, data, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smart_irrigation/websockets.py", line 197, in post
    await coordinator.async_update_zone_config(zone, data)
  File "/config/custom_components/smart_irrigation/__init__.py", line 1449, in async_update_zone_config
    await self._async_update_all()
  File "/config/custom_components/smart_irrigation/__init__.py", line 603, in _async_update_all
    weatherdata[const.RETRIEVED_AT] = datetime.datetime.now()
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object does not support item assignment
```

### How to reproduce

since 2024.6.3, already mentioned in issue https://github.com/jeroenterheerdt/HAsmartirrigation/issues/395#issue-2341473909

### Which version are you running?

2024.6.6

### Diagnostics file

- [X] I have attached a diagnostics file

### Additional information

[home-assistant_smart_irrigation_2024-06-10T19-00-14.710Z.log](https://github.com/user-attachments/files/15778064/home-assistant_smart_irrigation_2024-06-10T19-00-14.710Z.log)

**Comment by jeroenterheerdt** [comment]
yes, that is a bug fixed in v2024.6.7

**Comment by MrBurnst** [comment]
thanks! it works now with only sensors
